### DISCUSSION TITLE
Loading a lottie takes a long time on iPhone7, is there a way to optimize it?

### DISCUSSION THREAD

**Discussion by hardyZhi** [airbnb/lottie-ios#2174] [General]
Hello! II recently encountered a problem, loading a lottie took a long time on iPhone7 iOS15.7.8, is there a way to optimize it?

## Which Version of Lottie are you using?

Lottie 4.2.0

## Expected Behavior 
show soon

## Actual Behavior
It took several seconds to display and stuck the main thread

## Animation JSON
[rain.zip](https://github.com/airbnb/lottie-ios/files/12558156/rain.zip)
here the json

**Comment by hardyZhi** [comment]
if let lottieFilePath = jsonURL {
                DispatchQueue.main.async {
                    if let animation = LottieAnimation.filepath(lottieFilePath.path){
                        self.backLottieV.animation = animation
                        let lottieFilepathURl = lottieFilePath.deletingLastPathComponent().relativePath
                        let provider = FilepathImageProvider(filepath: lottieFilepathURl)
                        self.backLottieV.imageProvider = provider
                        self.backLottieV.loopMode = .repeat(MAXFLOAT)
                        self.backLottieV.backgroundBehavior = .pauseAndRestore
                        self.backLottieV.alpha = 1
                        self.backLottieV.play()
                    }else{
                        self.backLottieV.alpha = 0
                    }
                }
            }else{
                self.backLottieV.alpha = 0
            }

**Reply by calda** [reply]
Are you loading the `LottieAnimation` on a background thread and then dispatching back to the main thread? If not that's one good opportunity to improve things a bit.

**Reply by hardyZhi** [reply]
No, I only loaded it on the main thread. I took a look and found that the most time-consuming part was loading images.

**Comment by calda** [comment]
One area I want to investigate is moving this work to a background thread so it doesn't block the main thread. This should be possible, but we haven't tried it yet.

**Reply by hardyZhi** [reply]
The same lottie performs well on Android, but lags significantly on iOS.
